src/features/featureQuery.js

```diff
@@ -169,6 +169,13 @@ function chainingSubstitutionFormat3(contextParams, subtable) {
                         const substitution = lookup(glyphIndex);
                         if (substitution) substitutions.push(substitution);
                     }
+                } else if (substitutionType === '21') {
+                    for (let n = 0; n < inputLookups.length; n++) {
+                        const glyphIndex = contextParams.get(n);
+                        const substitution = lookup(glyphIndex);
+                        if (substitution)
+                            substitutions.concat(substitution);
```

Comment:
`.concat()` returns a new array, which is not stored anywhere here? So this should cause problems with specific substitutions.
We also want to avoid `.concat()` for performance reasons (see #690).

Reply:
Thank you for your reminder. I readed #690, you are right. I'll fix this.

Reply:
let me resolve this conversation once I did a full review